### ISSUE TITLE
Tokenizer Training

### ISSUE THREAD

**Issue by yukiarimo** [OpenMOSS/MOSS-TTS-Nano#54]
Hello!

Please release script for training the audio tokenizer!

Thanks!

**Comment by alpacaking** [comment]
Hello, thanks for your interest!

We plan to open-source the audio tokenizer training code in the future. Please stay tuned.

For MOSS-TTS-Nano, the current `MOSS-Audio-Tokenizer-Nano` is already sufficient for this TTS model. Re-training the audio tokenizer is unlikely to bring noticeable improvement for the current Nano TTS pipeline.

**Comment by yukiarimo** [comment]
I need to train just the decoder. My voice is OOD data, tokenizer is fully unable to reconstruct it perfectly (like my custom VITS-2 can); thus, speaker cannot match!

I just need loss functions and a training, cause when if will try to make one, it will get worse!

**Comment by alpacaking** [comment]
> I need to train just the decoder. My voice is OOD data, tokenizer is fully unable to reconstruct it perfectly (like my custom VITS-2 can); thus, speaker cannot match!
> 
> I just need loss functions and a training, cause when if will try to make one, it will get worse!

Thanks for the follow-up.

We are curious about the specific OOD audio case you mentioned. If you can share the audio sample, we would really appreciate it. That would help us better understand whether our current tokenizer has limitations on that type of data.

**Comment by yukiarimo** [comment]
No, I will NOT share the sample, because people will STEAL it and use it for TRAINING DATA.

The voice is in range of ~350 Hz fundamental frequency (female).

I have tested ALL tokenizers in the open-source space (no inference, just encode-decode the audio), and:

1. 90% of them are NOT 48 kHz, which I, as a musician DO hear the difference especially because my voice actress has HIGH f0
2. Even though ones that are 44.1/48 kHz do NOT match. I listened and compared spectrally via Audacity—the only model that matches the audio EXACTLY is my own VITS-2 model (cause VAE was trained on this voice, unlike tokenizers that just approximate) trained from scratch with updated configuration parameters for 48 kHz and long chunks.

By the way, if you know architectures that sounds better than VITS-2 that is:

1. Still phoneme-level only
2. Trainable from scratch on ~30 hours of audio
3. The ENTIRE architecture is trainable
4. Does one-step generation (aka NOT diffusion)

Please let me know!

Also, I’m planning on distilling the WavLM. Do you think 100 hours is enough?
